src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -896,11 +910,45 @@ private static bool AddFlags(TypeSymbol type, ArrayBuilder<bool> transformFlagsB
 
             private static void HandleCustomModifiers(int customModifiersCount, ArrayBuilder<bool> transformFlagsBuilder)
             {
-                for (int i = 0; i < customModifiersCount; i++)
+                // Native compiler encodes an extra transforms flag, always false, for each custom modifier.
+                transformFlagsBuilder.AddMany(false, customModifiersCount);
+            }
+        }
+
+        internal static class NativeIntegerTransformsEncoder
+        {
+            internal static ImmutableArray<TypedConstant> Encode(TypeSymbol type, TypeSymbol booleanType)
+            {
+                var builder = ArrayBuilder<bool>.GetInstance();
+                type.VisitType((typeSymbol, builder, isNested) => AddFlags(typeSymbol, builder, isNested), builder);
+                Debug.Assert(builder.Any());
+                Debug.Assert(builder.Contains(true));
+
+                var result = builder.SelectAsArray((flag, constantType) => new TypedConstant(constantType, TypedConstantKind.Primitive, flag), booleanType);
+                builder.Free();
+                return result;
+            }
+
+            private static bool AddFlags(TypeSymbol type, ArrayBuilder<bool> builder, bool isNestedNamedType)
+            {
+                switch (type.TypeKind)
                 {
-                    // Native compiler encodes an extra transforms flag, always false, for each custom modifier.
-                    transformFlagsBuilder.Add(false);
+                    case TypeKind.Array:
+                    case TypeKind.Pointer:
+                        builder.Add(false);
+                        break;
+
+                    default:
```

Comment:
>default [](start = 20, length = 7)

It is probably better to explicitly handle all kinds we know today and throw/assert from default case.